lib/elixir/pages/mix-and-otp/supervisor-and-application.md

```diff
@@ -213,50 +135,119 @@ The first step is to tell our application definition (for example, our `.app` fi
 
-Now that you have defined an application callback which starts our supervisor, we expect the `KV.Registry` process to be up and running as soon as we start `iex -S mix`. Let's give it another try:
+```elixir
+  def start(_type, _args) do
+    Registry.start_link(name: KV, keys: :unique)
+    Supervisor.start_link([], strategy: :one_for_one)
+  end
+```
+
+However, this would not be a good idea. In Elixir, we typically start processes inside supervision trees. In fact, we rarely use the `start_link` functions to start processes (except at the root of the supervision tree itself). Instead, do this:
 
 ```elixir
-iex> KV.Registry.create(KV.Registry, "shopping")
+  def start(_type, _args) do
+    children = [
+      {Registry, name: KV, keys: :unique}
+    ]
+
+    Supervisor.start_link(children, strategy: :one_for_one)
+  end
+```
+
+A supervisor receives one or more child specifications that tell exactly how to start each child. A child specification is typically represented by `{module, options}` pair, as shown above, and often as simply the module name. Sometimes, these children are supervisors themselves, giving ussupervision trees.
+
+Let's take it for a spin and see if we can indeed name our buckets using our new registry. Let's make sure to start a new `iex -S mix` (`recompile()` is not enough, as it does not reload your supervision tree) and then:
+
+```iex
+iex> name = {:via, Registry, {KV, "shopping"}}
+iex> KV.Bucket.start_link(name: name)
+{:ok, #PID<0.43.0>}
+iex> KV.Bucket.put(name, "milk", 1)
 :ok
-iex> KV.Registry.lookup(KV.Registry, "shopping")
-{:ok, #PID<0.88.0>}
+iex> KV.Bucket.get(name, "milk")
+1
 ```
 
-Let's recap what is happening. Whenever we invoke `iex -S mix`, it automatically starts our application by calling `Application.start(:kv)`, which then invokes the application callback. The application callback's job is to start a **supervision tree**. Right now, our supervisor has a single child named `KV.Registry`, started with name `KV.Registry`. Our supervisor could have other children, and some of these children could be their own supervisors with their own children, leading to the so-called supervision trees.
+Perfect, this time we didn't need to start the registry inside `iex`, as it was started as part of the application itself.
+
+By starting processes inside supervisors, we gain important properties such as:
+
+  * **Introspection**: for each application, you can fully introspect and visualize each process in its supervision tree, its memory usage, message queue, etc
+
+  * **Resilience**: when a process fails for an unexpected reason, its supervisor controls if and how those processes should be restarted, leading to self-healing systems
+
+  * **Graceful shutdown**: when your application is shutting down, the children of a supervision tree is terminated in the opposite order it was started, leading to graceful shutdowns
```

Comment:
```suggestion
  * **Graceful shutdown**: when your application is shutting down, the children of a supervision tree are terminated in the opposite order it was started, leading to graceful shutdowns
```

Reply:
since it's multiple **children**, probably also "in the opposite order they were started"?